api/v1alpha1/envoyproxy_types.go

```diff
@@ -127,6 +127,15 @@ type EnvoyProxyKubernetesProvider struct {
 	// +kubebuilder:validation:XValidation:message="loadBalancerIP can only be set for LoadBalancer type",rule="!has(self.loadBalancerIP) || self.type == 'LoadBalancer'"
 	// +kubebuilder:validation:XValidation:message="loadBalancerIP must be a valid IPv4 address",rule="!has(self.loadBalancerIP) || self.loadBalancerIP.matches(r\"^(((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\\.|$)){4})\")"
 	EnvoyService *KubernetesServiceSpec `json:"envoyService,omitempty"`
+
+	// EnvoyHpa defines the Horizontal Pod Autoscaler settings for Envoy Proxy Deployment.
+	// Once the HPA is being set, Replicas field from EnvoyDeployment will be ignored.
+	//
+	// +optional
+	// +kubebuilder:validation:XValidation:message="minReplicas must be greater than 0",rule="!has(self.minReplicas) || self.minReplicas > 0"
+	// +kubebuilder:validation:XValidation:message="maxReplicas must be greater than 0",rule="!has(self.maxReplicas) || self.maxReplicas > 0"
+	// +kubebuilder:validation:XValidation:message="maxReplicas cannot be less than minReplicas",rule="!has(self.minReplicas) || self.maxReplicas >= self.minReplicas"
```

Comment:
Should maxReplicas > self.minReplicas ?

Reply:
this is up to EG implementation, as per the Kubernetes HPA, it allows `maxReplicas == minReplicas`, hence I used `maxReplicas >= minReplicas`, thought @zhaohuabing ?

Reply:
Are there any specific cases where maxReplicas should be equal to mixReplicas? If no, this restriction should be enforced.

Reply:
How about making maxReplicas > self.minReplicas?  I doubt there won't be any cases requiring that these two be the same value, but we can always relax this restriction later if any users report that they need it.

Reply:
agreed